### ISSUE TITLE
1 failed deployment in a director should not prevent all deployment scraping

### ISSUE THREAD

**Issue by poblin-orange** [cloudfoundry-community/bosh_exporter#13]
Met this issue multiple time. previously it was with bosh director giving Http:500 error.

Here is what i get in the log

```
prometheus-server/2f817aa3-d74c-4dd4-a046-2455eba61cc1:/var/vcap/sys/log/bosh_exporter# tail -f *
==> bosh_exporter.stderr.log <==
time="2017-08-01T16:42:21Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607737' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T16:47:22Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607754' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T16:52:22Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607770' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T16:57:22Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607789' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:02:21Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607802' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:07:22Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607819' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:12:22Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607834' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:17:21Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607850' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:22:21Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607866' to succeed but was state is 'error'" source="bosh_collector.go:167"
time="2017-08-01T17:27:38Z" level=error msg="Error while reading Instances for deployment `logsearch`: Listing deployment 'logsearch' instances infos: Expected task '607882' to succeed but was state is 'error'" source="bosh_collector.go:167"

==> bosh_exporter.stdout.log <==
Content-Length: 232
Content-Range: bytes 0-231/232
Content-Type: text/plain;charset=utf-8
Date: Tue, 01 Aug 2017 17:27:38 GMT
Last-Modified: Tue, 01 Aug 2017 17:27:38 GMT
Server: nginx
Www-Authenticate: Basic realm="BOSH Director"
X-Content-Type-Options: nosniff

{"time":1501608458,"error":{"code":450001,"message":"Action Failed get_state: Getting processes status: Getting service status: Unmarshalling Monit status: read tcp 127.0.0.1:51630->127.0.0.1:2822: read: connection reset by peer"}}

```

I can reproduce manually the issue with that deployment with the bosh cli.

The issue i raise is that for this single stucked deployment, bosh_exporter doesnt export anything for the other sane deployments.
Would it make sense for bosh_exporter to isolate (ignore ?) this type of failed depls ?

**Comment by frodenas** [comment]
This is fixed at [v2.4.3](https://github.com/cloudfoundry-community/bosh_exporter/releases/tag/v2.4.3).

**Comment by poblin-orange** [comment]
thks @frodenas
